src/app/screens/swap/useAlexSponsoredTransaction.ts

```diff
@@ -19,11 +14,9 @@ const useAlexSponsorSwapEnabledQuery = (alexSDK: AlexSDK) =>
 export const useAlexSponsoredTransaction = (userOverrideSponsorValue: boolean) => {
   const alexSDK = useRef(new AlexSDK()).current;
   const [isServiceRunning, setIsServiceRunning] = useState(false);
-  const [hasPendingTransactions, setHasPendingTransactions] = useState(false);
   const { error, data: isEnabled, isLoading } = useAlexSponsorSwapEnabledQuery(alexSDK);
-  const { stxAddress } = useWalletSelector();
-  const selectedNetwork = useNetworkSelector();
-  const { data: stxPendingTxData } = useStxPendingTxData();
+  const { stxNonce } = useWalletSelector();
+  let hasPendingTransactions = false;
```

Comment:
should keep this initialisation together with the logic. (move this line down to L35)